http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ProxyService.scala

```diff
@@ -105,28 +107,39 @@ class ProxyService(
   final val requestTimeout = toScalaDuration(system.settings.config.getDuration("akka.http.server.request-timeout"))
   logger.info(s"Leo proxy request timeout is $requestTimeout")
 
-  private[leonardo] def getUserInfo(token: String)(implicit ev: Ask[IO, TraceId]): IO[(UserInfo, Instant)] =
-    for {
-      now <- IO.realTimeInstant
-      userInfo <- googleOauth2Service.getUserInfoFromToken(token)
-      samUserInfo <- samDAO.getSamUserInfo(token)
-      _ <- IO.fromOption(samUserInfo.map(_.userSubjectId))(AuthenticationError(Some(userInfo.userEmail)))
-    } yield (userInfo, now.plusSeconds(userInfo.tokenExpiresIn.toInt))
+  private[leonardo] def getUserInfo(token: String,
+                                    now: Instant)(implicit ev: Ask[IO, TraceId]): IO[(UserInfo, Instant)] =
+    decodeB2cToken(token, now) match {
+      case Left(_: JWTDecodeException) =>
+        for {
+          userInfo <- googleOauth2Service.getUserInfoFromToken(token)
+          samUserInfo <- samDAO.getSamUserInfo(token)
+          _ <- IO.fromOption(samUserInfo.map(_.userSubjectId))(AuthenticationError(Some(userInfo.userEmail)))
```

Comment:
Is the point of calling Sam here to check user enablement? If so then maybe should should `samUserInfo.enabled` instead..

Reply:
Then again I don't think we really need to check user enablement explicitly in `ProxyService`, since the proxy calls Sam `/api` routes to check resource permissions, which does the user enablement check as part of it. 

So could we just remove the `samDAO.getSamUserInfo` call?

Reply:
that sounds good to me

Reply:
So we can't really remove `samDAO.getSamUserInfo` from everywhere becuz `invalidateToken` and `setCookie` APIs also rely on this. I added a `checkUserEnabled` flag instead and we're skipping the Sam check only for proxy requests